test/functional/ui/embed_spec.lua

```diff
@@ -133,3 +136,51 @@ describe('--embed UI', function()
     ]]}
   end)
 end)
+
+describe('--embed --listen UI', function()
+  it('waits for connection', function()
+    clear ()
+    local child_server = assert(helpers.new_pipename())
+
+    -- attach immediately after startup, for early UI
+    local screen = thelpers.screen_setup(0,
+      string.format(
+        [=[["%s", "--embed", "--listen", "%s", "-u", "NONE", "-i", "NONE", "--cmd", "%s"]]=],
+        helpers.nvim_prog, child_server, helpers.nvim_set))
+    --screen.rpc_async = true  -- Avoid hanging. #24888
```

Comment:
Remove this line. If it hangs below what should be done is to change the `:request()` to `:notify()` for `nvim_ui_attach`.

Reply:
removed the rpc_async comment